swan-lake/development-tutorials/deployment-guide/serverless-deployment.md

```diff
@@ -0,0 +1,227 @@
+- [AWS Lambda - DynamoDB trigger](https://ballerina.io/learn/by-example/aws-lambda-dynamodb-trigger/)
+
+### CI/CD deployment with AWS Lambda
+
+This GitHub Action workflow automates the continuous integration and continuous deployment (CI/CD) process for a Ballerina project. 
+It triggers every push to the repository, and builds the project.
+```yaml
+name: Ballerina CI/CD with AWS Lambda
+
+on: [push]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the Repository
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Step 2: Build the Ballerina Project
+      - name: Ballerina Build
+        uses: ballerina-platform/ballerina-action@master
+        with:
+          args: build --cloud=aws_lambda
+
+      # Step 3: Package the Function for AWS Lambda
+      - name: Package for AWS Lambda
+        run: |
+          mkdir lambda
+          cp target/bin/your_project_name.jar lambda/
+          cd lambda
+          zip function.zip your_project_name.jar
+
+      # Step 4: Configure AWS Credentials
+        # AWS credentials are automatically picked up from GitHub Secrets
+
+      # Step 5: Deploy to AWS Lambda
+      - name: Deploy to AWS Lambda
+        uses: appleboy/lambda-action@master
+        with:
+          aws_access_key_id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws_region: ${{ secrets.AWS_REGION }}
+          function_name: your_lambda_function_name
+          handler_name: your_project_name.handler
+          runtime: java11
+          zip_file: lambda/function.zip
+          role: ${{ secrets.AWS_LAMBDA_ROLE_ARN }}
+          timeout: 30
+          memory_size: 512
+
+      # Optional: Test the Lambda Function
+      - name: Invoke Lambda Function
+        run: |
+          aws lambda invoke --function-name your_lambda_function_name --payload '{}' response.json
+          cat response.json
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+```
+- **Step 01:** The push event can initiate a build and deployment process
+- **Step 02:** This command tells Ballerina to compile the project specifically for AWS Lambda. It optimizes the output for Lambda’s serverless runtime requirements.
+- **Step 03:** Package the compiled Ballerina application into a ZIP file suitable for AWS Lambda deployment.
+- **Step 04:** Ensures that the workflow has the necessary credentials to interact with AWS services.
+- **Step 05:** Deploy the packaged Ballerina function to AWS Lambda by uploading the ZIP file.
+- **Step 06:** Test the deployed Lambda function by invoking it and outputting the response to confirm that the function works as expected.
+
+See [AWS Lambda](https://ballerina.io/learn/aws-lambda/) for more information.
```

Comment:
This is referred from multiple place and burried inside the text. So can we add the aws-lambda guide and azure-funtion guide links to the bottom of this page and that will help to find the links easily